ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java

```diff
@@ -284,7 +284,7 @@ public final int doPhase2Abort ()
 
 		org.omg.CosTransactions.Status s = get_status();
 
-		if ((s == org.omg.CosTransactions.Status.StatusCommitted)
+		if ((s == org.omg.CosTransactions.Status.StatusCommitted && preparedList.size() == 0)
```

Comment:
This kind of makes sense - if there was something in the intentions list but the transaction has already committed then have another go at completing it to resolve the intentions list (the else part of this test) by changing the state back to aborting (instead of committed). But this kind of implies we lied the first time around when we reported the transaction as committed. So my question is, are we certain that it is safe to morph the state back to a prior state (ie COMMITTED -> ABORTING)?